client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -52,6 +53,9 @@ const CancelPurchaseButton = React.createClass( {
 						}
 					}
 				), { persistent: true } );
+
+				clearPurchases();
```

Comment:
Good idea - this looked like a `wpcom` call to me at a glance because of the callback. I'll update it.